src/raftstore/store/worker/region.rs

```diff
@@ -266,13 +282,128 @@ impl SnapContext {
                 escape(&end_key),
                 e
             );
+        } else {
+            info!(
+                "succeed in deleting data in [{}, {})",
+                escape(&start_key),
+                escape(&end_key),
+            );
+        }
+    }
+}
+
+struct PendingDeleteRanges {
+    ranges: BTreeMap<Vec<u8>, (Vec<u8>, time::Instant)>,
+}
+
+enum RangeUpdateType {
+    TrimLeft = 0,
+    TrimRight = 1,
+    TrimMiddle = 2,
+    Discard = 3,
+}
+
+impl PendingDeleteRanges {
+    // find out the ranges that need to be trimed or discarded
+    fn find_overlap(
+        &self,
+        start_key: Vec<u8>,
+        end_key: Vec<u8>,
+        ranges: &mut Vec<(RangeUpdateType, Vec<u8>)>,
```

Comment:
`find_overlap`'s only duty is to find overlapped ranges, don't add update type into it.